core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -113,6 +116,18 @@
 
     private List<String> services = new ArrayList<>();
 
+    private Set<ExposedService> exposedServices = new HashSet<>();
+
+    @Value
+    class ExposedService {
```

Comment:
Note that `@EqualsAndHashCode` will include `waitStrategy` into the hashcode, so `exposedServices` will end up with **two** entries for the same name/port.

Let's add `@EqualsAndHashCode` explicitly and set its `of` to either `name` or `name` + `port` (whichever makes sense)